### PR TITLE
Implement adding and removing of book

### PR DESCRIPTION
**Pull request by yusufnviiri**
- Configure components.
- Add relevant hooks to the project.
- install and use redux tools
- Implement add and remove of books
-

### CODE REVIEW

File: src/components/BookInput.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { ACTIONS } from '../redux/books/books';
+
+function BookInput() {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [type, setType] = useState('');
+
+  // console.log(books);
```
Comment (Davidosky007) [inline src/components/BookInput.js]: * [OPTIONAL] kindly remove all `commented codes` instances from your codebase as it makes your code base bulky and unprofessional

File: src/redux/books/actions.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable import/prefer-default-export */
```
Comment (Davidosky007) [inline src/redux/books/actions.js]: * [OPTIONAL] Kindly try to fix linter errors rather than inline disabling, as linters help us write bug-free code

**Review by Davidosky007 — APPROVED** [review]
Hi @yusufnviiri 👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


nice-job
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 

100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+export function booksReducer(state = initial(), action) {
+  switch (action.type) {
+    case ACTIONS.ADD_BOOK:
+      return [...state, addBook(action.payload.type, action.payload.title, action.payload.author)];
+    case ACTIONS.REMOVE_BOOK:
+      return state.filter((book) => book.id !== action.payload.id);
+    default:
+      return state;
+  }
+}
```
Comment (mateo951) [inline src/redux/books/books.js:29]: Well done 🔥 state is immutable in all reducers ✔

File: src/index.js [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
 import App from './App';
+import store from './redux/configureStore';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
   <React.StrictMode>
-    <BrowserRouter>
-      <App />
-    </BrowserRouter>
+    <Provider store={store}>
```
Comment (mateo951) [inline src/index.js:12]: Main React component is wrapped in Provider and assign store 🙌

File: src/components/BookContainer.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  Routes,
+  Route,
+} from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import BookList from './BookList';
+import BookInput from './BookInput';
+import Category from './Category';
+
+function BookContainer() {
+  const books = useSelector((state) => state.books);
```
Comment (mateo951) [inline src/components/BookContainer.js:12]: List of books displays books from Redux store (by using `useSelector` function) 💪😬

File: src/components/BookInput.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+          />
+          <input
+            type="text"
+            name="type"
+            value={type}
+            onChange={(e) => setType(e.target.value)}
+            placeholder="add type"
+          />
+
+          <button type="submit">ADD BOOK</button>
```
Comment (mateo951) [inline src/components/BookInput.js:47]: Nice! A button (or a form) for adding new books 😉

**Review by mateo951 — APPROVED** [review]
Hi @yusufnviiri, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🙌 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookInput.js (hunk 6 above)
Comment (yusufnviiri) [reply src/components/BookInput.js:47]: thanks @mateo951 and @Davidosky007  ….your comments are  mesmerizing🙏🙏🙏

**Pull request merged**